src/main/java/mate/jdbc/service/CarServiceImpl.java

```diff
@@ -0,0 +1,76 @@
+package mate.jdbc.service;
+
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+import mate.jdbc.dao.CarDao;
+import mate.jdbc.exception.DataProcessingException;
+import mate.jdbc.lib.Inject;
+import mate.jdbc.lib.Service;
+import mate.jdbc.model.Car;
+import mate.jdbc.model.Driver;
+
+@Service
+public class CarServiceImpl implements CarService {
+    @Inject
+    private CarDao carDao;
+
+    @Override
+    public Car create(Car car) {
+        return carDao.create(car);
+    }
+
+    @Override
+    public Car get(Long id) {
+        return carDao.get(id)
+                .orElseThrow(() -> new DataProcessingException("Could not get car from DAO "
+                + "by id = " + id));
+    }
+
+    @Override
+    public List<Car> getAll() {
+        return carDao.getAll();
+    }
+
+    @Override
+    public Car update(Car car) {
+        return carDao.update(car);
+    }
+
+    @Override
+    public boolean delete(Long id) {
+        return carDao.delete(id);
+    }
+
+    @Override
+    public void addDriverToCar(Driver driver, Car car) {
+        if (car.getDrivers() != null) {
+            car.getDrivers().add(driver);
+            update(car);
+            return;
+        }
+        Set<Driver> drivers = new HashSet<>();
+        drivers.add(driver);
+        car.setDrivers(drivers);
+        update(car);
+    }
+
+    @Override
+    public void removeDriverFromCar(Driver driver, Car car) {
+        if (car.getDrivers() != null && !car.getDrivers().isEmpty()) {
+            car.getDrivers().remove(driver);
+            update(car);
+            return;
+        }
+        throw new RuntimeException("Can't remove driver from '"
+                + car.getManufacturer().getName() + " " + car.getModel() + "'!"
+                + System.lineSeparator()
+                + " You have not yet added any drivers to '"
+                + car.getManufacturer().getName() + " " + car.getModel() + "'!");
```

Comment:
```suggestion
		car.getDrivers().remove(driver);
		update(car);
```

I think we can just remove driver and update car. In this case, the business logic is built in such a way that a situation in which we try to remove a driver from a car that has no drivers is impossible

Reply:
Fixed.